### ISSUE TITLE
Resource hangs if it doesn't exist when trying to delete

### ISSUE THREAD

**Issue by kungfoome** [crossplane-contrib/provider-pagerduty#9]
### What happened?
If you try to delete the manifest for a resource and you delete it manually, it will hang and never delete the manifest. You have to patch the finalizers to actually remove the manifest.

I get this error 
`Warning  CannotObserveExternalResource  31s (x4 over 9m31s)  managed/escalation.pagerduty.crossplane.io/v1alpha1, kind=policy  cannot run refresh: refresh failed:`

### How can we reproduce it?
Create an escalation policy with the crossplane pagerduty resource
Go to pagerduty and manually delete the escalation policy
Try and remove the escalation policy manifest from k8s and wait. It will just hang and do nothing

### What environment did it happen in?
Crossplane version: 
Provider version: 0.3.0

**Comment by haarchri** [comment]
Can you add your manifest ? I can have a look

**Comment by kungfoome** [comment]
Yes, I was just using this example here https://github.com/crossplane-contrib/provider-pagerduty/blob/main/examples-generated/service/service.yaml

I created all these things and the reassigned the service to another policy within PD manually. Then removed the policy manually as well within PD. Went back and tried to remove the resource from kubernetes and that is what i got.
